crypto/merkle/simple_proof.go

```diff
@@ -67,7 +66,8 @@ func SimpleProofsFromMap(m map[string][]byte) (rootHash []byte, proofs map[strin
 
 // Verify that the SimpleProof proves the root hash.
 // Check sp.Index/sp.Total manually if needed
-func (sp *SimpleProof) Verify(rootHash []byte, leafHash []byte) error {
+func (sp *SimpleProof) Verify(rootHash []byte, leaf []byte) error {
```

Comment:
Did we change the semantics of Verify here? It used to take the hash of the thing passed to SimpleMerkle, and now it takes the pre-image bytes?

Reply:
Yeah, we had to so that everything that uses it doesn't have to be aware of the 0 byte prefix.